validator/src/journal/chain.rs

```diff
@@ -87,8 +88,13 @@ pub trait ExternalLock: Send + Sync {
     fn lock(&self) -> Box<ExternalLockGuard>;
 }
 
-pub trait ExternalLockGuard: Drop {
-    fn release(&self);
+pub trait ExternalLockGuard {
+    fn notify_on_chain_updated(
+        &self,
+        chain_head: Option<BlockWrapper>,
+        committed_batches: Vec<Batch>,
+        uncommitted_batches: Vec<Batch>,
+    );
```

Comment:
I believe you can remove the ExternalLock and ExternalLockGuard traits now

Reply:
Yeah, I think that is correct.